src/fuzzer.ts

```diff
@@ -0,0 +1,111 @@
+import random from "crypto-random-bigint";
+import { InferType, IntType, SourceUnit, TypeNode, FunctionDefinition } from "solc-typed-ast";
+import { encodeFunctionCall } from "web3-eth-abi";
+import { runEVM, RunEVMResult } from "./evmc";
+
+export type FuzzingResult = {
+    runResult: RunEVMResult;
+    version: string;
+    encodedCall: string;
+    callParameters: number[];
+    functionName: string;
+};
+
+export class Fuzzer {
+    private functionCall: string;
+    private testFunction: FunctionDefinition;
+    private callParameters: number[];
+
+    constructor(
+        private unit: SourceUnit,
+        private testCallFunction: string,
+        private version: string
+    ) {
+        const testFunction = this.unit.vContracts[0].vFunctions.find(
+            (f) => f.name === this.testCallFunction
+        );
+        if (testFunction === undefined) {
+            throw new Error(`Function ${this.testCallFunction} not found in contract`);
+        }
+        this.testFunction = testFunction;
+        this.callParameters = this.generateCallParameters();
+        this.functionCall = this.prepareFunctionCall();
+    }
+
+    private generateCallParameters(): number[] {
+        const type = new InferType(this.version);
+        const params = [];
+        for (const param of this.testFunction.vParameters.vParameters) {
+            const typeNode = type.variableDeclarationToTypeNode(param);
+            const value = this.generateRandomValue(typeNode);
+            if (value !== undefined) {
+                // assert the value is defined
```

Comment:
Looks like this needs to be an assert?

Reply:
Fixed in [93d139d](https://github.com/ConsenSysDiligence/solc-fuzz/pull/3/commits/93d139d70ab85ce749f0f582f9f3e8ef16784f5c)